store/sqlstore/channel_store.go

```diff
@@ -2302,6 +2302,104 @@ func (s SqlChannelStore) performSearch(searchQuery string, term string, paramete
 	return result
 }
 
+func (s SqlChannelStore) getSearchGroupChannelsQuery(userId, term string, isPostgreSQL bool) (string, map[string]interface{}) {
+	var query, baseLikeClause string
+	if isPostgreSQL {
+		baseLikeClause = "ARRAY_TO_STRING(ARRAY_AGG(u.Username), ', ') LIKE %s"
+		query = `
+            SELECT
+                *
+            FROM
+                Channels
+            WHERE
+                Id IN (
+                    SELECT
+                        cc.Id
+                    FROM (
+                        SELECT
+                            c.Id
+                        FROM
+                            Channels c
+                        JOIN
+                            ChannelMembers cm on c.Id = cm.ChannelId
+                        JOIN
+                            Users u on u.Id = cm.UserId
+                        WHERE
+                            c.Type = 'G'
+                        AND
+                            u.Username = (SELECT Username FROM Users WHERE Id = :UserId)
+                        GROUP BY
+                            c.Id
+                    ) cc
+                    JOIN
+                        ChannelMembers cm on cc.Id = cm.ChannelId
+                    JOIN
+                        Users u on u.Id = cm.UserId
+                    GROUP BY
+                        cc.Id
+                    HAVING
+                        %s
+                    LIMIT
+                        ` + strconv.Itoa(model.CHANNEL_SEARCH_DEFAULT_LIMIT) + `
+                )`
+	} else {
+		baseLikeClause = "GROUP_CONCAT(u.Username SEPARATOR ', ') LIKE %s"
```

Comment:
I'm a little concerned about the performance of this but if we add elasticsearch for enterprise installations it probably won't be an issue

Reply:
As discussed, it's not clear the impact this fix would have in the performance as per the query and it being just part of the More Direct Channels modal, and adding elasticsearch support to it would increase its scope. We will wait to gather more feedback and decide what the next step is.

Reply:
I think it should be more or less performance, the most inner subquery restrict the query to the Group channels of the user, so... you can be talking about maybe 1000 (for example), and that means at top 7000 member rows, so... I don't see a huge deal, but for sure we need to check the performance at least for that example. I would suggest to create an environment with enough users to create 7000 group chat with one user, and then, check the performance of that call